capability.go

```diff
@@ -0,0 +1,128 @@
+package versionbundle
+
+import (
+	"encoding/json"
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+)
+
+// Capability represents the data format being exposed by authorities like
+// microservices or operators.
+type Capability struct {
```

Comment:
Naming. /shrug

I feel you. I heard this feedback already but so far I did not get a better alternative suggestion. 

This structure is the structure an authority exposes. So our operators expose this structure to explain to the outside what it is able to do. So there has to be some container that explains which authority the version information is about, plus a list of "bundles" which describe which components and dependencies the authority provides in which combination. `Capability` was the best I could came up with. Feel free to suggest something better.

Reply:
I put more thought into it. What do you think about adding `Name` field to a bundle and get rid of `Capability` completely? That way we can have something `/version-bundles` endpoint which would return array of `Bundle`.

That also would make `Aggregation` struct contain `Bundles [][]Bundle` field which makes sense. IMO then it also makes sense to rename `Aggregation` to `Distributions` `, which is collection of software components built, assembled and configured so that it can be used essentially "as is"`.

I know you wrote:

> I would like to keep aggregation here since it describes what it is based on the process 

I think the product is more interesting than the process creating it. Another nice thing is you can change the process and product name remains. When I call an API I'm interested in a product. E.g. I don't want to change path from `/aggregations` to `/reductions` because the process changed.

Reply:
I tried already having a list and having a redundant name in the bundle. I found that odd because you have to set the same name in multiple bundles. I thought this is weird as well and opted against it. 

```
[
  {
    ...
    Name: "authority",
    ...
  },
  {
    ...
    Name: "authority",
    ...
  },
  {
    ...
    Name: "authority",
    ...
  },
]
```

Having this made more sense to me. 

```
{
  Bundles: [
    {
      ...
    },
    {
      ...
    },
    {
      ...
    },
  ],
  Name: "authority",
}
```

This is why there is an "extra container" called `Capability`. Would you be fine with having the redundant name in the bundles?I get your picture and I like the idea to have a list of distributions at the end where one distribution is a list of bundles. It is then probably rather `[]Distribution` instead of `[][]Bundle` because the distribution will have to have an identifier and maybe other properties.

Reply:
> Would you be fine with having the redundant name in the bundles?

Yes.

> []Distribution instead of [][]Bundle

I agree. Distribution type which has `[]Bundle` field and additional metadata is nice.